### ISSUE TITLE
False positives of `UselessAccessModifier` for blocks inside class

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#7467]
Hello!

There are false positives of `UselessAccessModifier` for blocks inside class.

--------

## Expected behavior

No offenses.

## Actual behavior

There is offense for the second `private` from `UselessAccessModifier`.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

## CLI
class CLI < Clamp::Command
  subcommand 'foo' do
    def execute
      make_foo
    end

    private

    def make_foo; end
  end

  subcommand 'bar' do
    def execute
      make_bar
    end

    private

    def make_bar; end
  end
end
```

## RuboCop version

```
> rubocop -V
0.75.1 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-linux)
```

**Comment by koic** [comment]
Can you specify the following `ContextCreatingMethods` option in your .rubocop.yml?

```yaml
# .rubocop.yml
Lint/UselessAccessModifier:
  ContextCreatingMethods:
    - subcommand
```

https://rubocop.readthedocs.io/en/stable/cops_lint/#lintuselessaccessmodifier

**Comment by koic** [comment]
And I opened PR #7471 to update document for the cop.

**Comment by AlexWayfer** [comment]
> Can you specify the following ContextCreatingMethods option in your .rubocop.yml?

Yes, I can. Didn't see it. My bad.

> And I opened PR #7471 to update document for the cop.

Thank you!
